### ISSUE TITLE
Import from trello

### ISSUE THREAD

**Issue by brantje** [nextcloud/deck#30]
Hi there!
Awesome idea, currently i use trello.
It would be a pain to migrate by hand, however, trello offers a [API](https://developers.trello.com/advanced-reference).
Would it be possible to write some kind of importer?

**Comment by juliushaertl** [comment]
That would be definitely super nice to have also for other apps, but it is no priority for me now. 

We could also offer a public API for deck so that people can build simple import scripts for various other tools similar to this one: https://github.com/matueranet/kanboard-import-trello

**Comment by srbaker** [comment]
This is what I need the API for, so I can import from Trello. Happy to do the work, as I mentioned in #83.

**Comment by weipah** [comment]
I +1 this feature request. It would really ease up the process to make the switch from cloud-based to local installation.

Trello also supports JSON exports in their free version.

I guess using this for a one-time import would be much easier than deep-diving into their api.

**Comment by juliushaertl** [comment]
@weipah The API was basically meant to be for Deck so users can script their import from various platforms. 

I see the sense of importing trello data directly from the webinterface, but I will not have time to implement this in the near future. Several other features will be focused before. But as always, if anyone wants to contribute, pullrequests are always welcome.

**Comment by krombel** [comment]
can this be reopened again? I would like to have that feature as well.

**Comment by juliushaertl** [comment]
This would be a nice feature, however currently there are no plans to implement this in the near future. In general I would be a lot more happy if importing from various 3rd party services and software would be done using the API, so we keep the deck code clean from other dependencies. Thus I will keep this ticket closed for now. If somebody wants to implement this feature nevertheless we are happy to assist and help out.

If you wish to have this feature implemented by the Nextcloud GmbH there is the option for consulting work on top of your Nextcloud Enterprise subscription to get your features implemented.

**Comment by BloodyIron** [comment]
Let's get this reopened please. I'm for it!

**Comment by martinaware** [comment]
I'd like it to reopened as well.

**Comment by brokencode64** [comment]
Would love to see this re-opened. Currently this is the only thing keeping me from migrating to Deck.

**Comment by putt1ck** [comment]
@brokencode64 we definitely have this on our radar but will be via the API using Trello export JSON format (because last I checked that was only free export format).

**Comment by juliushaertl** [comment]
> If somebody wants to implement this feature nevertheless we are happy to assist and help out.

This still counts even if the ticket is closed. 

**A general note:**
Please use the [Github Reactions](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments) feature instead of commenting that also need a specific feature. That way we can see your support without notifying all users subscribed to a given issue.

Now on the trello import:
Imo the first step would be to have a general import/export available (#87), once that is in place we can thing about how the trello json could be mapped to our data structures.

**Comment by maxammann** [comment]
I'm currently working on writing an external import tool. I have now an in-memory view of the trello data which can be imported into Deck.

**Comment by BloodyIron** [comment]
What about Trello boards that have Epics and other big aspects that Deck doesn't yet have?

**Comment by maxammann** [comment]
@BloodyIron so far I did not encounter Epics in the JSON export of Trello. PowerUps etc will be lost I suppose. The migration has to be lossy.

Checklists are appened as markdown in the description for example.

**Comment by stefan-niedermann** [comment]
@maxammann maybe you can post the link to your repository here? Then this kind of questions can be handeled in separate issues.

> PowerUps etc will be lost I suppose

Maybe there will be a way in the future to export PowerUp stuff as `json` and attach it as a `json` file attachment to the cards or so...

**Comment by maxammann** [comment]
There you go: I'll create issues for missing stuff: https://github.com/maxammann/trello-to-deck
